### PR TITLE
[SPARK-44048][DOCS] Remove sql-migration-old.md

### PR DESCRIPTION
**Pull request by wangyum**
### What changes were proposed in this pull request?

This PR removes sql-migration-old.md.

### Why are the changes needed?

We can use sql-migration-guide.md directly without redirection:
image

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

manual test.

### CODE REVIEW

File: docs/_data/menu-sql.yaml [hunk 1]
```diff
@@ -71,7 +71,7 @@
 - text: PySpark Usage Guide for Pandas with Apache Arrow
   url: sql-pyspark-pandas-with-arrow.html
 - text: Migration Guide
-  url: sql-migration-old.html
+  url: sql-migration-guide.html
```
Comment (dongjoon-hyun) [inline docs/_data/menu-sql.yaml:74]: Do you happen to know the history?

Comment (wangyum) [reply docs/_data/menu-sql.yaml:74]: cc @HyukjinKwon

Comment (HyukjinKwon) [reply docs/_data/menu-sql.yaml:74]: It was to keep the old URL so existing links are not broken. I think it's been pretty a while so I think we can just remove.

**Review by HyukjinKwon — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM, too.

**Comment by dongjoon-hyun** [comment]
According to the affected version info of JIRA, I landed it to `master` branch only.

**Pull request closed without merging**